src/scheduler/task.rs

```diff
@@ -334,6 +334,53 @@ impl PriorityTaskQueue {
 			IDLE_PRIO
 		}
 	}
+
+	/// Change priority of specifc task
+	pub fn set_priority(&mut self, handle: TaskHandle, prio: Priority) -> Result<(), ()> {
+		let i = handle.get_priority().into() as usize;
+		let mut pos = self.queues[i].head.as_mut().ok_or(())?;
+
+		loop {
+			if handle.id == pos.borrow().id {
+				let task = pos.clone();
+
+				{
```

Comment:
Let's add a comment for clarity:

```suggestion
				// Extract found task from queue and set new priority
				{
```